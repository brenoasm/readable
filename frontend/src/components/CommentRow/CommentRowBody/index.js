import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import Ratings from '../../Ratings';

import { colors } from '../../../theme';
import { UP_VOTE, DOWN_VOTE } from '../../../utils/ratings';

const propTypes = {
  comment: PropTypes.object,
  modifyCommentVoteValues: PropTypes.func
};

const defaultProps = {
  comment: {},
  modifyCommentVoteValues: () => {}
};

const StyledCommentRowBody = styled.div`
  display: flex;
  height: 100px;
  margin-top: 10px;

  > div:first-child {
    border-radius: 5px
    padding: 8px;
    overflow-y: auto;
    box-shadow: 0 3px 5px ${colors.primary.primaryFour};
    background-color: ${colors.white};
    flex-grow: 10;

    p {
      line-height: 0.5;
    }
  }

  > div:last-child {
    text-align: center;
    height: 100%;
    flex-grow: 1;
    display: flex;
    flex-direction: column;
    justify-content: center;
  }
`;

const CommentRowBody = ({
  comment,
  modifyCommentVoteValues
}) => (
  <StyledCommentRowBody>
    <div>
      <p>{comment.body}</p>
    </div>
    <div>
      <Ratings
        voteScore={comment.voteScore}
        rateUp={() => modifyCommentVoteValues(comment, UP_VOTE)}
        rateDown={() => modifyCommentVoteValues(comment, DOWN_VOTE)}
      />
    </div>
  </StyledCommentRowBody>
);

CommentRowBody.propTypes = propTypes;
CommentRowBody.defaultProps = defaultProps;

export default CommentRowBody;
