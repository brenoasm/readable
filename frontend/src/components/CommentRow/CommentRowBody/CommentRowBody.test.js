/* eslint-disable */

import React from 'react';

import CommentRowBody from './';

import CommentMock from '../../../testHelpers/mocks/comments';

describe('CommentRowbody', () => {
  let wrapper;

  beforeEach(() => {
    const initialState = {
      comment: CommentMock.comments[0],
      modifyCommentVoteValues: jest.fn()
    };

    wrapper = shallow(<CommentRowBody {...initialState} />);
  });

  afterEach(() => {
    wrapper.unmount;
  });

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have one Ratings component', () => {
    expect(wrapper.find('Ratings')).toHaveLength(1);
  });

});
