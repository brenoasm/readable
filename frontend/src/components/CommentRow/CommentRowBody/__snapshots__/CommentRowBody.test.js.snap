// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentRowbody matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CommentRowBody
    comment={
        Object {
            "author": "Breno",
            "body": "Test",
            "deleted": false,
            "id": "1",
            "parentDeleted": false,
            "parentId": "1",
            "timestamp": 1538417732043,
            "voteScore": 1,
          }
    }
    modifyCommentVoteValues={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <div>
          <p>
                    Test
          </p>
</div>,
        <div>
          <Ratings
                    rateDown={[Function]}
                    rateUp={[Function]}
                    voteScore={1}
          />
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <p>
            Test
</p>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Test",
          },
          "ref": null,
          "rendered": "Test",
          "type": "p",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Ratings
            rateDown={[Function]}
            rateUp={[Function]}
            voteScore={1}
/>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "rateDown": [Function],
            "rateUp": [Function],
            "voteScore": 1,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <div>
            <p>
                        Test
            </p>
</div>,
          <div>
            <Ratings
                        rateDown={[Function]}
                        rateUp={[Function]}
                        voteScore={1}
            />
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <p>
              Test
</p>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Test",
            },
            "ref": null,
            "rendered": "Test",
            "type": "p",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Ratings
              rateDown={[Function]}
              rateUp={[Function]}
              voteScore={1}
/>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "rateDown": [Function],
              "rateUp": [Function],
              "voteScore": 1,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
